codalab/model/bundle_model.py

```diff
@@ -915,7 +915,15 @@ def transition_bundle_preparing(self, bundle, user_id, worker_id, start_time, re
 
             bundle_update = {
                 'state': State.PREPARING,
-                'metadata': {'started': start_time, 'last_updated': start_time, 'remote': remote},
+                'metadata': {
+                    'started': start_time,
+                    'last_updated': start_time,
+                    'remote': remote,
+                    'remotes': (bundle.metadata.remotes or [])
```

Comment:
`remotes` is defined as previous, which suggests it shouldn't include `remote`.  Perhaps this should be named `previous_remotes`?

Reply:
Renamed it to `remote_history` instead. It should include the current remote as well.